h2o-r/h2o-package/R/automl.R

```diff
@@ -266,15 +267,20 @@ h2o.automl <- function(x, y, training_frame,
     finally = if (is_progress) h2o.show_progress()
   )
 
-  leaderboard[,2:length(leaderboard)] <- as.numeric(leaderboard[,2:length(leaderboard)])  # Convert metrics to numeric
-  # If leaderboard is empty, create a "dummy" leader
-  if (nrow(leaderboard) > 1) {
-      leader <- h2o.getModel(automl_job$leaderboard$models[[1]]$name)
+  # If the leaderboard is empty, it creates a dummy row so let's remove it
+  if (leaderboard$model_id[1,1] == "") {
+    leaderboard <- leaderboard[-1,]
+    warning("The leaderboard contains zero models: try running AutoML for longer (the default is 1 hour).")
+  }
+  # If leaderboard is not empty, grab the leader model, otherwise create a "dummy" leader
+  if (nrow(leaderboard) > 0) {
+    leaderboard[,2:length(leaderboard)] <- as.numeric(leaderboard[,2:length(leaderboard)])  # Convert metrics to numeric
+    leader <- h2o.getModel(automl_job$leaderboard$models[[1]]$name)
   } else {
-      # create a phony leader
-      Class <- paste0("H2OBinomialModel")
-      leader <- .newH2OModel(Class = Class,
-                             model_id = "dummy")
+    # create a phony leader
+    Class <- paste0("H2OBinomialModel")
+    leader <- .newH2OModel(Class = Class,
+                           model_id = "dummy")
```

Comment:
what prevents you from reusing `h2o.getAutoML` directly from L254? 
Would be better than having to duplicate code:
```
aml <- h2o.getAutoML(res$job$dest$name)
aml$project_name <- build_control$project_name
return(aml)
```

Reply:
@seb-h2o Yeah you're right, I can just do that.  Thanks!

Reply:
@seb-h2o I changed the `h2o.automl()` function to make use of `h2o.getAutoML()`. Does everything else look good?